### DISCUSSION TITLE
404 acme-challenge

### DISCUSSION THREAD

**Discussion by artooras** [vercel/vercel#8336] [Help]
I keep getting the following 404 error for my app hosted on Vercel:

`404 GET [my_app_domain]/.well-known/acme-challenge/[hash]`

The full error is as follows:

```
{
   "dt":"2022-08-06 14:08:16.554 UTC",
   "message_string":"START RequestId: d876c4af-e78a-4eca-bf03-c6bdf52c34bb Version: $LATEST\nEND RequestId: d876c4af-e78a-4eca-bf03-c6bdf52c34bb\nREPORT RequestId: d876c4af-e78a-4eca-bf03-c6bdf52c34bb\tDuration: 584.99 ms\tBilled Duration: 585 ms\tMemory Size: 1024 MB\tMax Memory Used: 143 MB\tInit Duration: 1158.19 ms\t\n",
   "platform_string":"Vercel",
   "vercel":{
      "deployment_id_string":"dpl_FTfof3FyJZTqtsD3gLcb3yWmL37k",
      "host_string":[my_host_string],
      "id_string":"1659794896561278405679953085",
      "path_string":"_error",
      "project_id_string":"prj_exoTV9cnO7o0IxnGuylrDeMO0kbA",
      "proxy":{
         "client_ip_string":"74.125.210.17",
         "host_string":[my_host],
         "method_string":"GET",
         "path_string":"/.well-known/acme-challenge/wBHeC9B201wJUsFdKq-ezE2XMA7oWXaEftC8bBCEcc0",
         "region_string":"iad1",
         "scheme_string":"https",
         "status_code_integer":"404",
         "timestamp_integer":"1659794896554",
         "user_agent_string":"[\"Google-Firebase\"]",
         "vercelcache_string":"MISS"
      },
      "request_id_string":"iad1::iad1::crm6n-1659794896554-cbfd7f2c1530",
      "source_string":"lambda",
      "source_type_string":"http",
      "statuscode_integer":"404"
   }
}
```

I'm not sure under what circumstance these occur, but they are quite frequent.

Based on [Vercel documentation](https://vercel.com/blog/automatic-ssl-with-vercel-lets-encrypt), I understand that this is related to the provisioning of the SSL certificate.

Is there perhaps something I need to do on my end in terms of configuration?

**Comment by amyegan** [comment]
Hi @artooras. Sorry to hear you are running into this error. Are you seeing this error consistently, or is it an intermittent problem? The user-agent string is Google Firebase so I'm wondering if you are using Firebase with your Project and have a custom domain set up across platforms. If so, something there might be misconfigured or may just need to be reset. 

Please share any more details you can to help me investigate. If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would be even better.

**Reply by artooras** [reply]
Hi @amyegan . It's difficult to identify a pattern for these errors, but it seems that it's rather periodic. The `acme-challenge` requests with 404 errors are getting logged every 15-60 minutes or so. Is there an automated process from Vercel's side for these pings perhaps?

**Reply by amyegan** [reply]
What do you get back if you run this command?
```
curl http://example.com/.well-known/acme-challenge -I
```

If you're using something like Cloudflare proxy, there may be some configuration adjustments to get it all running smoothly: https://vercel.com/guides/using-cloudflare-with-vercel

**Reply by artooras** [reply]
I am indeed using Cloudflare, but as DNS-only, without proxy. And I do have the CNAME record with `cname.vercel-dns.com`.
If I run the command you mentioned, I do get the `HTTP/1.1 404 Not Found` as per the description in the linked page.
